src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -0,0 +1,179 @@
+package org.broadinstitute.consent.http.service;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch.core.BulkRequest;
+import co.elastic.clients.elasticsearch.core.BulkResponse;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.broadinstitute.consent.http.configurations.ElasticSearchConfiguration;
+import org.broadinstitute.consent.http.db.DacDAO;
+import org.broadinstitute.consent.http.db.DataAccessRequestDAO;
+import org.broadinstitute.consent.http.db.DatasetDAO;
+import org.broadinstitute.consent.http.db.UserDAO;
+import org.broadinstitute.consent.http.models.Dataset;
+import org.broadinstitute.consent.http.models.DatasetProperty;
+import org.broadinstitute.consent.http.models.Study;
+import org.broadinstitute.consent.http.models.StudyProperty;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.elastic_search.DatasetTerm;
+import org.broadinstitute.consent.http.models.elastic_search.StudyTerm;
+import org.broadinstitute.consent.http.models.elastic_search.UserTerm;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+
+public class ElasticSearchService implements ConsentLogger {
+
+  private final ElasticsearchClient esClient;
+  private final ElasticSearchConfiguration esConfig;
+  private final DatasetDAO datasetDAO;
+  private final DataAccessRequestDAO dataAccessRequestDAO;
+  private final DacDAO dacDAO;
+  private final UserDAO userDAO;
+  private final UseRestrictionConverter useRestrictionConverter;
+
+  public ElasticSearchService(
+      ElasticsearchClient esClient,
+      ElasticSearchConfiguration esConfig,
+      UseRestrictionConverter useRestrictionConverter,
+      DatasetDAO datasetDAO,
+      DataAccessRequestDAO dataAccessRequestDAO,
+      DacDAO dacDAO,
+      UserDAO userDAO) {
+    this.esClient = esClient;
+    this.esConfig = esConfig;
+    this.useRestrictionConverter = useRestrictionConverter;
+    this.datasetDAO = datasetDAO;
+    this.dataAccessRequestDAO = dataAccessRequestDAO;
+    this.dacDAO = dacDAO;
+    this.userDAO = userDAO;
+  }
+
+  public BulkResponse indexDatasets(List<DatasetTerm> datasets) throws IOException {
+    BulkRequest.Builder br = new BulkRequest.Builder();
+
+    for (DatasetTerm dataset : datasets) {
+      br.operations(op -> op
+          .index(idx -> idx
+              .index(esConfig.getDatasetIndexName())
+              .id(dataset.getDatasetId().toString())
+              .document(dataset)
+          )
+      );
+    }
+
+    return esClient.bulk(br.build());
```

Comment:
With all of the null checks handled, I was able to get to this statement. It ended up failing with about 600+ terms
```
{
  "message": "[es/bulk] failed: [action_request_validation_exception] Validation Failed: 1: type is missing;
...
```
It also took about 4 minutes to complete the execution (run against all datasets) so we should probably look into batching up groups of datasets as opposed to trying to do them all in one go.

Reply:
The failure message leads me to think there's an incompatibility between the most recent java library's bulk call and our very old ES server instance (see this [SO](https://stackoverflow.com/questions/73446339/elasticsearch-exception-type-action-request-validation-exception-reason-valida) post). 

In the past, due to these kinds of inherent issues, I've relied on the Http client and not the java client. It might make sense to try that approach here.